documentation/deliberately_vulnerable_flask_app/integration_test_utils.py

```diff
@@ -0,0 +1,158 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from __future__ import annotations
+
+import json
+import logging
+import subprocess
+import sys
+from pathlib import Path
+from typing import final
+
+LOG: logging.Logger = logging.getLogger(__name__)
+
+
+@final
+class PyreErrorException(Exception):
+    """
+    Custom Exception to raise when Pyre errors out
+    """
+
+    pass
+
+
+def normalized_json_dump(
+    results: str, strip_locations: bool, filter_issues: bool
+) -> str:
+    """
+    Returns a normalised JSON string from results keeping only essential items. Filters
+    issues down to issues that have the code we intend to test for if filter_issues is
+    true, and removes location information if strip_location is set to True.
+    """
+    normalized = json.loads(results)
+    if "errors" in normalized:
+        pretty_error = json.dumps(normalized, sort_keys=True, indent=2)
+        raise PyreErrorException(
+            f"Errors were found when processing analysis results:\n{pretty_error}"
+        )
+
+    if filter_issues:
+        # Filter down to only issues that have the code that we intended to test for.
+        # This prevents the introduction of new rules or false positives from
+        # breaking existing tests.
+        normalized = [
+            issue for issue in normalized if f"test_{issue['code']}_" in issue["define"]
+        ]
```

Comment:
That seems copy pasted from somewhere. Out of curiosity, where did you get this?

I don't think we need this here.

Reply:
Yes it's from the Instagram integration tests the idea is to have common functions shared between our internal integration tests and the Opensource